### ISSUE TITLE
AutoPkgr creating new package entry for each update

### ISSUE THREAD

**Issue by FabianN** [lindegroup/autopkgr#542]
Hi,

I'm having an odd issue that I at first thought was only affecting packages with an override, but now know it also affects packages I have not created overrides for.

The effect that I am seeing:

Whenever a package has an update and AutoPkgr downloads the new version (done via schedule), instead of updating the current package entry, a new package entry is created.

For example, I had an entry for AdobeFlashPlayer with version 21.something.something downloaded with all the current settings that it's recipie has. AdobeFlashPlayer version 22.something.something comes out and a new entry is created, using all the settings given in it's recipe. For the clients to pull the newest version of AdobeFlashPlayer I need to go into the Catalog and remove the old version and enable the new version in the Catalogs that I have that package enabled for.

I have seen this effect with most of the packages that I have created overrides for, such as AdobeFlashPlayer, Firefox, Google Chrome, Onyx, Skype, and VLC. I have also seen this effect with the munkitools2 package that does not have any override (though, I did at one point create an override and then later Removed the Override). There are a couple that have not had updates since Munki was setup and so I can't confirm that they have the same issue. But all packages that have updated after the initial package entry was created have had duplicate entries created.

I am doing all of my munki management through these programs (using GUI front-ends to manage it all):
- Munki Tools 2.7.1.2753
- AutoPkg 0.6.1
- AutoPkgr 1.4.1
- MunkiAdmin 1.4.2

The process I used to create or remove recipe overrides is done through AutoPkgr

This is the contents of ~/Library/Application Support/AutoPkgr/recipie_list.txt:

```
com.github.autopkg.munki.munkitools2
com.github.rtrouton.install.AutoPkg-Release
local.munki.VLC
local.munki.TextWrangler
local.munki.Synergy
local.munki.SourceTree
local.munki.Skype
local.munki.OracleJava8
local.munki.9.Onyx10
local.munki.11.Onyx10
local.munki.10.Onyx10
local.munki.MSOffice2011Updates
local.munki.GoogleChrome
local.munki.AdobeReaderDC
local.munki.AdobeFlashPlayer
local.munki.Firefox
MakeCatalogs.munki
```

Let me know if there's any other information I can provide to help hunt this issue down.

**Comment by homebysix** [comment]
Hi @FabianN,

It's expected behavior for a new version to appear in your Munki repo (instead of overwriting the previous version, as you seem to imply is your desired behavior). Going in and testing the package and adding it to the stable catalog is part of the normal workflow.

So I guess I'm not sure that I see the problem here. Are you getting multiple duplicates of the _same_ version of software imported into your Munki repo? If so, that would be a problem worth troubleshooting.

**Comment by roodavis** [comment]
I am getting multiple duplicates of the same version of software imported into my repo by AutoPkgr at one location. What would cause this and/or how would I trouble shoot it?

**Comment by homebysix** [comment]
Hi @roodavis - Although it's not uncommon for the same version of software to download on subsequent runs ([see the first FAQ here](https://github.com/autopkg/autopkg/wiki/FAQ)), it is not expected behavior for the same version of software to import into your Munki repo multiple times.

Can you take a look at your ~/Library/Application Support/AutoPkgr/recipe_list.txt file and make sure it contains MakeCatalogs.munki at the end?

And can you try running this command 2-3 times in a row to see whether the duplication persists?
```
/usr/bin/python /usr/local/bin/autopkg run --verbose --recipe-list ~/Library/Application\ Support/AutoPkgr/recipe_list.txt
```

Thanks!

**Comment by roodavis** [comment]
Hi @homebysix -- It's only been a few months, but I finally got around to looking at this issue again. Today was the first day back after spring break so I checked MunkiAdmin to see if there were any new downloads I needed to approve and had 13 copies of Google Chrome, Firefox, and Shotcut. Several additional apps/recipes were just fine. After deleting all the extras I decided to search for a solution again and saw your comments. I ran the above command twice and I see comments about the package already exists and the downloaded packages removed from the Cache, the command ends with "Nothing downloaded, packaged or imported." I also still get the warning about missing trust info and fail recipes not set. So I am thinking there is some setting or permission in AutoPkgr that is causing it to read permissions incorrectly allowing multiple copies to be downloaded.

I have this same setup at a smaller school with neither of the above issues. This installation has 2000+ clients, so I am hesitant to start the whole configuration over again. But that may be a summer project.
